language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs

```diff
@@ -149,7 +149,10 @@ data Token a = LexName { loc        :: a
              deriving (Show, Eq, Generic, NFData)
 
 asBytes :: Word8 -> Doc a
-asBytes = Text 2 . T.pack . ($ mempty) . showHex
+asBytes x = Text 2 $ T.pack $ addLeadingZero $ ($ mempty) $ showHex x
```

Comment:
If you're going to make it point-ful at least get rid of the weird `($ mempty)` thing.